src/protocols/rdp/channels/audio-input/audio-buffer.c

```diff
@@ -242,6 +338,7 @@ void guac_rdp_audio_buffer_write(guac_rdp_audio_buffer* audio_buffer,
         pthread_mutex_unlock(&(audio_buffer->lock));
         return;
     }
+    pthread_mutex_unlock(&(audio_buffer->lock));
```

Comment:
This mutex is documented as guarding all access to `audio_buffer`. Unlocking things here and then operating on the `audio_buffer` without holding the mutex looks like a threadsafety issue.

If you are certain this is safe, can you describe how that has been verified?